phabricatoremails/mail.py

```diff
@@ -87,17 +87,23 @@ def to_mime_message(self, from_address, include_target_in_subject=False):
         thread_index_base = thread_id_md5.encode("utf-8")[:27]
         thread_index = thread_index_base + b" " + struct.pack(">i", self.timestamp)
 
+        return {
+            "Thread-Topic": thread_id,
+            "Thread-Index": base64.encodebytes(thread_index).decode("utf-8"),
+            "In-Reply-To": thread_message_id,
+            "References": thread_message_id,
+        }
+
+    def to_mime_message(self, from_address, include_target_in_subject=False):
         msg = MIMEMultipart("alternative")
-        msg["Thread-Topic"] = thread_id
-        msg["Thread-Index"] = base64.encodebytes(thread_index).decode("utf-8")
-        msg["In-Reply-To"] = thread_message_id
-        msg["References"] = thread_message_id
         msg["From"] = self.encode_from(from_address)
         msg["To"] = self.to
         msg["Date"] = formatdate(timeval=self.timestamp)
         msg["Subject"] = (
             f"|{self.to}| {self.subject}" if include_target_in_subject else self.subject
         )
+        for name, value in self._generate_threading_headers(from_address).items():
+            msg[name] = value
```

Comment:
nit:

```suggestion
        msg.update(self._generate_threading_headers(from_address))
```

Reply:
Can you do `update()` on non-dictionaries? Unsure if it's viable with [`MIMEMultipart`](https://docs.python.org/3/library/email.mime.html#email.mime.nonmultipart.MIMENonMultipart), but I'll take a closer look.

Reply:
Ah fair point, it looks like you can not do that -- please disregard.